Us9 1 create swap

CECDO opened the pull request:
creation of the swap form

CECDO committed:
creation swap entity

CECDO committed:
creation of the swap form

CECDO committed:
Merge branch 'dev' into US9_1_createSwap

CECDO committed:
Delete Version20220112140644.php

CECDO committed:
Update User.php

RomainClair committed:
Suppress PHPMD warnings for User entity

Bruno-Fernandes-37 reviewed (changes requested):
overriding methods that should not be in swapController, no version file ..... no big deal for sure

Benjamin-51 committed:
Recreation of another migration file.

Benjamin-51 committed:
Fix for the two methods that were overwritten.

I am not sure what happened since no commit shows that the file was deleted after creating this branch from dev.

Benjamin-51 committed:
Add a link for making a swap.

RomainClair reviewed (changes requested):
The fixtures need to be fixed according to the database changes

Benjamin-51 commented:
> The fixtures need to be fixed according to the database changes

What fixtures would we need to change? We are unsure about if this current PR requires changes or if we need to update them like adding skills and comments to our users.

Commit:
add notation=3 in user entity

Benjamin-51 committed:
Merge branch 'dev' into US9_1_createSwap

Bruno-Fernandes-37 reviewed (dismissed):
it all looks good for me

Commit:
set skill to swap form according to route

Commit:
Merge branch 'dev' into US9_2_createSwap

Benjamin-51 committed:
Merge branch 'US9_2_createSwap' of https://github.com/WildCodeSchool/purple-php-2109-project3-skillswap into US9_1_createSwap

RomainClair commented on src/Controller/SwapController.php:
```diff
@@ -34,4 +42,44 @@ public function display(Skill $skill, User $user): Response
+                $swap->setAsker($currentUser);
+                $swap->setHelper($helper);
+                $swap->setSkill($skill);
+                $swap->setDate(new DateTime());
```
This could have been done in the Swap construct

RomainClair commented on src/Entity/Swap.php:
```diff
@@ -0,0 +1,149 @@
+     */
+    private \DateTimeInterface $date;
+
+    /**
+     * @ORM\Column(type="text")
+     */
+    private string $message;
```
You probably should have constraints to validate the message... May be avoiding empty messages ?
